### PR TITLE
REL-1197: Rdf export/import from neo4j

### PR DESCRIPTION
**Pull request by vbyndych**
## Goal
Rdf export/import from neo4j

## Changelog
- feat: ontology import/export for neo4j.
- fix: removed duplicated properties.

### CODE REVIEW

**Review by yaraslau-kavaliou — APPROVED** [review]

File: core/kernel/persistence/starsql/class.StarRdf.php [hunk 1]
```diff
@@ -60,97 +43,117 @@ protected function getPersistence()
...
+
+        $query  = null;
+        if (!empty($systemSubjectList)) {
+            $systemNode = Query::node('Resource');
+            $query = Query::new()->match($systemNode)
+                ->where($systemNode->property('uri')->in(array_keys($systemSubjectList)))
+                ->set($systemNode->labeled('System'));
+        }
+
+        return Statement::create($query->build());
```
Comment (augustas) [inline core/kernel/persistence/starsql/class.StarRdf.php:157]: If `$systemSubjectList` is empty, then `$query` remains to be `null`, and this line errors out

File: core/kernel/persistence/starsql/class.StarIterator.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+use oat\generis\model\data\Model;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use oat\generis\model\kernel\persistence\starsql\FlatRecursiveIterator;
+
+/**
+ * Iterator over all triples
+ *
+ * @author joel bout <joel@taotesting.com>
+ * @package generis
```
Comment (augustas) [inline core/kernel/persistence/starsql/class.StarIterator.php:30]:
```suggestion
```

File: core/kernel/persistence/starsql/class.StarIterator.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2002-2008 (original work) 2014 Open Assessment Technologies SA
```
Comment (augustas) [inline core/kernel/persistence/starsql/class.StarIterator.php:18]:
```suggestion
 * Copyright (c) 2023 (original work) Open Assessment Technologies SA;
```

File: core/kernel/persistence/starsql/class.StarRdf.php (hunk 1 above)
Comment (vbyndych) [reply core/kernel/persistence/starsql/class.StarRdf.php]: thank you good catch

File: core/kernel/persistence/starsql/class.StarIterator.php (hunk 2 above)
Comment (vbyndych) [reply core/kernel/persistence/starsql/class.StarIterator.php]: fixed

File: core/kernel/persistence/starsql/class.StarIterator.php (hunk 3 above)
Comment (vbyndych) [reply core/kernel/persistence/starsql/class.StarIterator.php]: fixed

**Review by augustas — APPROVED** [review]

**Pull request merged**